### PR TITLE
libpod: resolve volume paths per entry when copying archive to stopped container

### PR DESCRIPTION
**Pull request by shivaansh0610-LUFFY**
#### What this PR does

Updates `copyFromArchive` when copying a tar archive into a stopped container that has mounted volumes or bind mounts.
Previously, calling `PUT /containers/{id}/archive?path=/` on a stopped container resolved the destination root strictly to the container's rootfs layer. `copier.Put` extracted entries targeting `/mymount/...` directly into the underlying rootfs storage layer instead of the volume storage directory (`_data`). When the container subsequently started, the volume was mounted over `/mymount`, hiding the extracted files.

This PR updates `copyFromArchive` for stopped containers with volumes/mounts to resolve the exact host storage destination (volume directory or bind mount source) per archive entry via `c.resolvePath()`. Also adds an API v2 regression test in `23-containersArchive.at`.

Related to: #21861

#### Checklist

Ensure you have completed the following checklist for your pull request to be reviewed:

- [x] I have read and understood our [contributing guidelines](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md) and will not have more than two open PRs as a new contributor.
- [x] PR description, commit message, and GitHub comments are human-written, per [LLM Policy](https://github.com/podman-container-tools/community/blob/main/LLM_POLICY.md)
- [x] Certify you wrote the patch or otherwise have the right to pass it on as an open-source patch by signing all
commits. (`git commit -s`). (If needed, use `git commit -s --amend`).  The author email must match
the sign-off email address. See [CONTRIBUTING.md](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md#sign-your-prs)
for more information.
- [x] Referenced issues using `Fixes: #00000` in commit message (if applicable)
- [x] [Tests](https://github.com/podman-container-tools/podman/tree/main/test#readme) have been added/updated (or no tests are needed)
- [x] [Documentation](https://github.com/podman-container-tools/podman/blob/main/docs/README.md) has been updated (or no documentation changes are needed)
- [x] All commits pass `make validatepr` (format/lint checks)
- [x] [Release note](https://github.com/kubernetes/community/blob/master/contributors/guide/release-notes.md) entered in the section below (or `None` if no user-facing changes)

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: libpod/container_copy_common.go [hunk 1]
```diff
@@ -229,6 +235,54 @@ func (c *Container) copyFromArchive(path string, chown, noOverwriteDirNonDir boo
...
+		targetContainerPath := filepath.Join(path, hdr.Name)
+		entryRoot, entryPath, _, err := c.resolvePath(mountPoint, targetContainerPath)
+		if err != nil {
+			return err
+		}
+
+		targetDir := entryPath
+		if hdr.Typeflag != tar.TypeDir {
+			targetDir = filepath.Dir(entryPath)
+		}
```
Comment (giuseppe) [inline libpod/container_copy_common.go]: aren't directories created at the wrong level now?

File: test/apiv2/23-containersArchive.at [hunk 2]
```diff
@@ -82,4 +82,31 @@ t POST containers/${CTR}/exec $TMPD/exec.json 201 .Id~[0-9a-f]\\{64\\}
...
+podman volume create ${VOLNAME} &>/dev/null
+podman create --name ${CTR_VOL} -v ${VOLNAME}:/volmount "${IMAGE}" top &>/dev/null
+
+VOL_TMPD=$(mktemp -d podman-apiv2-test.volarchive.XXXXXXXX)
+mkdir -p ${VOL_TMPD}/volmount
+echo "volcontent" > ${VOL_TMPD}/volmount/volfile.txt
+VOL_TAR="${VOL_TMPD}/vol.tar"
+tar -C ${VOL_TMPD} -cvf ${VOL_TAR} volmount/volfile.txt &>/dev/null
+
+t PUT "/containers/${CTR_VOL}/archive?path=%2F" ${VOL_TAR} 200 ''
```
Comment (giuseppe) [inline test/apiv2/23-containersArchive.at:99]: please add a test also for directories and symlinks

**Review by giuseppe — COMMENTED** [review]
please squash the commits into one and use your real name and email for the git commit and the `Signed-off-by:` line

**Comment by shivaansh0610-LUFFY** [comment]
Thanks for the review @giuseppe!

1. Fixed directory level extraction by computing filepath.Rel(entryRoot, entryPath) so files, subdirectories, and symlinks extract at the exact right path.
2. Added tests for subdirectories and symlinks in 23-containersArchive.at.
3. Squashed everything into 1 commit signed off with my real name.

Ready for another look!

**Comment by giuseppe** [comment]
I still see two commits

**Comment by shivaansh0610-LUFFY** [comment]
> I still see two commits

Ah my bad! Just squashed it into 1 single commit now. Thanks @giuseppe!

**Comment by danishprakash** [comment]
> [x] I have read and understood our [contributing guidelines](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md) and will not have more than two open PRs as a new contributor.

JFYI

**Comment by shivaansh0610-LUFFY** [comment]
> > [x] I have read and understood our [contributing guidelines](https://github.com/podman-container-tools/podman/blob/main/CONTRIBUTING.md) and will not have more than two open PRs as a new contributor.
> 
> JFYI

Got it, thanks @danishprakash! I'll keep this PR in Draft until one of my open PRs gets merged.

**Review by Honny1 — CHANGES_REQUESTED** [review]
I did a quick check of CI results, and the failures are serious. Please fix CI before the next round of review.

**Comment by shivaansh0610-LUFFY** [comment]
@honny1 I've updated the PR with a clean implementation and addressed the initial feedback:

1. Host path bind-mounting for stopped containers:  mountContainerVolumesAndMounts() now bind-mounts all named volume storage directories (_data) and bind mount sources into mountPoint on the host before streaming the archive. This allows buildahCopiah.Put and Get to process the archive in a single pass while preserving hardlinks, PAX headers, and performance.

2. Volume state cleanup: Updated copyFromArchive() cleanup to set NeedsCopyUp = false and CopiedUp = true on all attached named volumes, preventing subsequent podman start calls from overwriting copied files.

3. Clean build & checks: ci / Validate source code changes, ci / Cross Build (Linux, FreeBSD), DCO, and unit/apiv2 tests are all passing cleanly.

I would really appreciate your guidance/help on how you'd prefer copyFromArchive to handle the remaining volume path resolution edge cases in the integration tests for stopped containers. Thanks!

**Comment by Honny1** [comment]
@shivaansh0610-LUFFY Can you please update the PR on the upstream main?

**Comment by Honny1** [comment]
@shivaansh0610-LUFFY, any update on this?

**Comment by shivaansh0610-LUFFY** [comment]
> @shivaansh0610-LUFFY, any update on this?

@Honny1 I will Update it by today.
